### ISSUE TITLE
[JQ] PRIORITY- retraining the classifier

### ISSUE THREAD

**Issue by RFOxbury** [nestauk/dap_job_quality#59]
Currently the model uses jobbert embedding layer and then inputs that into logistic regression.

Better options are:

- [x] Fine-tune jobbert (like company description classifier)
- [x] fine-tune smaller classification model (distilbert?)
- [x] Use faster embeddings (all-minilm) and input them into classifier (eg logistic regression again)

Other improvements are:

- [x] training on a balanced sample (previously, it was unbalanced - more 0s than 1s)
- [x] cluster the negative sentences and take a sample from each cluster, so that the negative sample gives good coverage of the type of negative text the algorithm might end up seeing
- [ ] padding out underrepresented dimensions of job quality with fictional examples generated by GPT?
    - [x] label positive samples (in Google Sheets) so that we know if any dimensions are over/underrepresented
    - [x] Find additional examples from underrepresented categories from `prinz-green-jobs/outputs/data/ojo_application/deduplicated_sample/ojo_sample.csv`
    - [x] Label additional examples as positive or negative

Cosmetic/practical improvements include:

- structuring the code in such a way that it's easy to move into ojd_daps_language_models later down the line - eg
    - [x] having hyperparameters in a config
    - [ ] using metaflow
